internal/heartbeat/heartbeat.go

```diff
@@ -85,6 +90,42 @@ func (s *HeartbeatData) RetrieveInfo() models.Heartbeat {
 	return heartbeatInfo
 }
 
+func (s *HeartbeatData) buildHardwareInfo() *models.HardwareInfo {
+	var hardwareInfo *models.HardwareInfo
+	var err error
+	if s.previousMutableHardwareInfo == nil {
+		// Only send all Hw info (mutable + immutable) for the 1st heartbeat, then send only mutable hw info
```

Comment:
Umm, what happens if heartbeat got 401/403 and there is no data submitted to the API, shouldn't this be on the send() function if everything is ok?

Reply:
What do you mean?

Reply:
you send the first Hearbeat(Full info) and operator returns 401. 
the next thing that you're going to send, it's a delta, operator will know about the full hardware?

Reply:
Indeed, I should add something to reset previous when handling error [here](https://github.com/project-flotta/flotta-device-worker/blob/c8ae0c9b7c8fb78f810514c3c8e30c2cf9eccf2f/internal/heartbeat/heartbeat.go#L160)   and [here](https://github.com/project-flotta/flotta-device-worker/blob/c8ae0c9b7c8fb78f810514c3c8e30c2cf9eccf2f/internal/heartbeat/heartbeat.go#L171)

And I should add one [here](https://github.com/project-flotta/flotta-device-worker/blob/c8ae0c9b7c8fb78f810514c3c8e30c2cf9eccf2f/internal/heartbeat/heartbeat.go#L176)

So I'll have to introduce a bool to reset previous only if the first send fails

Reply:
I added check [here](https://github.com/project-flotta/flotta-device-worker/blob/c8ae0c9b7c8fb78f810514c3c8e30c2cf9eccf2f/internal/heartbeat/heartbeat.go#L235) and I reset the previous if there is an error during the 1st heartbeat